sled-agent/src/rack_setup/service.rs

```diff
@@ -1363,14 +1370,34 @@ impl ServiceInner {
 
         // Wait until time is synchronized on all sleds before proceeding.
         rss_step.update(RssStep::WaitForTimeSync);
-        let sled_addresses: Vec<_> = sled_plan
-            .sleds
-            .values()
-            .map(|initialization_request| {
-                get_sled_address(initialization_request.body.subnet)
+        let ntp_addresses: Vec<_> = service_plan
+            .services
+            .iter()
+            .flat_map(|(_, sled_config)| {
```

Comment:
I'm kinda tempted to suggest we do something here to confirm we have exactly one NTP zone per sled (probably via asserting, since we came up with the plan and it should always be correct?). Maybe that would be overkill, since 0 or more-than-one NTP zone on a sled is going to cause problems pretty quickly anyway? Something like (using `exactly_one()` from itertools):

```rust
let ntp_addresses: Vec<_> = service_plan
    .services
    .iter()
    .map(|(_, sled_config)| {
        sled_config.zones.iter().filter_map(|zone_config| {
            // ...extract ntp_admin_addr...
        })
        .exactly_one()
        .expect("plan should specify one NTP zone per sled")
    })
    .collect();
```

Reply:
Sure thing, updated